frontend/src/pages/detail/PODetail.tsx

```diff
@@ -143,33 +143,41 @@ export default function PODetail() {
       header: "Book",
       customBody: (rowData: POPurchaseRow) =>
         booksDropDownEditor(rowData.bookTitle, (newValue) => {
-          rowData.bookTitle = newValue;
-          setBookDropdownRefreshKey(Math.random());
+          setPurchases((draft) => {
+            const purchase = findById(draft, rowData.id);
+            purchase!.bookTitle = newValue;
```

Comment:
Maybe add a boolean check for the purchase! statement to ensure a check for non-null value and if its null it throws an error.

Reply:
Normally I would agree, but in this scenario I think it's reasonable to use the non-null assertion operator (!). That's because this code is run when a specific row is selected (which is contained in rowData), and we know that there has to be a purchase in the purchases array that has this ID, or something in our code is seriously wrong (and we wouldn't really be able to safely do anything useful on this page)